sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -197,32 +197,55 @@ class AioHttpStreamDownloadGenerator(AsyncIterator):
 
     :param pipeline: The pipeline object
     :param response: The client response object.
-    :param block_size: block size of data sent over connection.
-    :type block_size: int
+    :param bool decompress: If True which is default, will attempt to decode the body based
+        on the ‘content-encoding’ header.
     """
-    def __init__(self, pipeline: Pipeline, response: AsyncHttpResponse) -> None:
+    def __init__(self, pipeline: Pipeline, response: AsyncHttpResponse, decompress: bool = True) -> None:
         self.pipeline = pipeline
         self.request = response.request
         self.response = response
         self.block_size = response.block_size
+        self._decompress = decompress
         self.content_length = int(response.internal_response.headers.get('Content-Length', 0))
-        self.downloaded = 0
 
     def __len__(self):
         return self.content_length
 
     async def __anext__(self):
+        if not(self._decompress) and self.pipeline:
+            try:
+                auto_decompress = self.pipeline.transport.session.auto_decompress
+                self.pipeline.transport.session.auto_decompress = False
```

Comment:
We should not mutate the session since the session can be shared. Which means that we shouldn't have to make the transport public either.

Reply:
Agree. But the auto_decompress in aiohttp is a per-session setting. If we don't support session mutation, we have to make it as a per-session setting as well. Your idea?

Reply:
Just throwing it out as an option; we can do the decompression outside of aiohttp and disable the decompression for the sessions we create. It may be tricky to do in a backwards compatible way, however.